setup.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+
+# type: ignore
+import os
+import shutil
+import subprocess
+from pathlib import Path
+
+import setuptools.command.build_py as orig
+from setuptools import setup
+
+LANGKIT = (
+    "git+https://github.com/AdaCore/langkit.git"
+    "@45df941275a0844fafc960cfc6b775b1108f42f3#egg=langkit"
+)
+
+
+base_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+# Hack: To be able to place the shared libraries into the Python package
+# (without setting LD_LIBRARY_PATH before starting the python interpreter)
+# we need patch the RPATH of the parser library. While the parse library
+# itself is loaded via dlopen() directly, the langkit_base library which
+# is placed in the same directory cannot be found. To fix that, we add
+# "$ORIGIN" to the RPATH of librecordfluxdsllang.so. The simplest way to
+# do this is the `patchelf` tool.
+def patch_rpath():
+    sopath = "lib/librecordfluxdsllang/relocatable/dev/librecordfluxdsllang.so"
+    rpath = subprocess.check_output(["patchelf", "--print-rpath", sopath])
+    subprocess.run(
+        ["patchelf", "--set-rpath", "$ORIGIN:" + rpath.decode("utf-8"), sopath], check=True
+    )
+
+
+# We cannot import langkit.* or language.* globally, as langkit needs to be installed first.
+def manage_factory():
+    # pylint: disable=import-outside-toplevel
+    from langkit.compile_context import CompileCtx
+    from langkit.libmanage import ManageScript
+
+    from language.lexer import rflx_lexer as lexer
+    from language.parser import rflx_grammar as grammar
+
+    class Manage(ManageScript):
+        def create_context(self, args):
+            return CompileCtx(lang_name="RecordFluxDSL", lexer=lexer, grammar=grammar)
+
+    return Manage()
+
+
+class BuildWithParser(orig.build_py):
+    def initialize_options(self):
+        os.environ["GNATCOLL_ICONV_OPT"] = "-v"
```

Comment:
We should check if this variable is set before and restore the original afterwards. This call alters the environment even after the script has finished.

Reply:
No, it doesn't:

```Console
$ export BAR=1
$ python3 -c 'import os; os.environ["BAR"] = "2"'
$ echo $BAR
1
$
```

The environment is only changed for the current process and (potentially) its child processes.